cmd/supernode/app/root.go

```diff
@@ -174,11 +175,11 @@ func setAdvertiseIP() error {
 	if err != nil {
 		return errors.Wrapf(errorType.ErrSystemError, "failed to get ip list: %v", err)
 	}
-	if len(ipList) != 0 {
-		cfg.AdvertiseIP = ipList[0]
+	if len(ipList) == 0 {
+		return errors.Wrapf(errorType.ErrSystemError, "get ip list empty")
```

Comment:
@lowzj  should we use `127.0.0.1` or returns error here? WDYT?

Reply:
As for Java version of Dragonfly, that's exactly what we did as a solution. And for Golang version, we should find a better way to set the IP of supernode to support for multiple networks or something else. And for now, we use either user-specified AdvertiseIP or the first non-loop address if the AdvertiseIP is empty as a temporary solution. WDYT?

Reply:
It can be set in the following order:
1. user-specified advertiseIP
2. first non-loop address
3. the parameter `superNodeIp` of the first register request from dfget

Reply:
I understand what you said. And I will do the third item in the pr #457. 
Just return here that I will do.

Reply:
Has been updated. PTAL. THX >